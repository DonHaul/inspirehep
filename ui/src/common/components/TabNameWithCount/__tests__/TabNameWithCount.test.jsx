import { render } from '@testing-library/react';

import TabNameWithCount from '../TabNameWithCount';

describe('TabNameWithCount', () => {
  it('renders with required props', () => {
    const { asFragment } = render(<TabNameWithCount name="Test" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders when loading true', () => {
    const { getByTestId } = render(<TabNameWithCount name="Test" loading />);
    expect(getByTestId('loading')).toBeInTheDocument();
  });

  it('does not display count if loading is true', () => {
    const { getByTestId, queryByText } = render(
      <TabNameWithCount name="Test" loading count={10} />
    );
    expect(getByTestId('loading')).toBeInTheDocument();
    expect(queryByText('(10)')).toBeNull();
  });

  it('displays count if loading is false', () => {
    const { getByText } = render(
      <TabNameWithCount name="Test" loading={false} count={10} />
    );
    expect(getByText('(10)')).toBeInTheDocument();
  });
});
