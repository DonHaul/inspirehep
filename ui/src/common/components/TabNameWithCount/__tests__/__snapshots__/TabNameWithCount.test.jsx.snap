// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabNameWithCount renders with required props 1`] = `
<DocumentFragment>
  <span>
    <span>
      Test
    </span>
    <span
      class="ml1"
    />
  </span>
</DocumentFragment>
`;
